app/helpers/remote_execution_helper.rb

```diff
@@ -77,12 +77,11 @@ def template_invocation_status(task)
     end
   end
 
-  def template_invocation_actions(task, host)
-    if task.nil?
-      []
-    else
-      [display_link_if_authorized(_("Details"), hash_for_template_invocation_path(:id => task).merge(:auth_object => host, :permission => :view_hosts))]
-    end
+  def template_invocation_actions(task, host, job_invocation)
+    [
+      display_link_if_authorized(_('View host'), hash_for_host_path(host).merge(:auth_object => host, :permission => :view_hosts)),
+      display_link_if_authorized(_('Rerun on %s') % host.name, hash_for_rerun_job_invocation_path(:id => job_invocation, :host_ids => [ host.id ])),
```

Comment:
I wanted to be clear about on which host it's going to be run, without following the whole line to the left. But I can easily change it to 'Rerun' only if you prefer. Or maybe 'Rerun on this host'